daffodil-tdml/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -437,69 +449,121 @@ abstract class TestCase(testCaseXML: NodeSeq, val parent: DFDLTestSuite)
+      }
     }
+    allImpls.foreach {
+      _.setValidateDFDLSchemas(parent.validateDFDLSchemas)
+    }
+    if (allImpls.isEmpty) throw new TDMLException("No TDML DFDL implementations found for '%s'".format(implementationStrings.mkString(", ")))
+    allImpls
   }
 
-  private def compileProcessor(schemaSource: DaffodilSchemaSource, useSerializedProcessor: Boolean): DFDLTestSuite.CompileResult = {
-    val pf = compiler.compileSource(schemaSource)
-    val diags = pf.getDiagnostics
-    if (pf.isError) {
-      Left(diags) // throw new TDMLException(diags)
-    } else {
-      val res = this.generateProcessor(pf, useSerializedProcessor)
-      res
-    }
+  private lazy val tdmlDFDLProcessorFactoryTraitName = {
+    import scala.language.reflectiveCalls
+    // Doing this so that if we decide to rename the verbose class name here
+    // it will handle this code too.
+    // Also we can rename the packages, etc.
+    val clazzTag = scala.reflect.classTag[TDMLDFDLProcessorFactory]
+    val cname = clazzTag.toString()
+    cname
   }
 
-  final protected def getProcessor(schemaSource: DaffodilSchemaSource, useSerializedProcessor: Boolean): DFDLTestSuite.CompileResult = {
-    val res: DFDLTestSuite.CompileResult = schemaSource match {
-      case uss: URISchemaSource =>
-        SchemaDataProcessorCache.compileAndCache(uss, useSerializedProcessor,
-          parent.checkAllTopLevel,
-          root,
-          null) {
-            compileProcessor(uss, useSerializedProcessor)
-          }
-      case _ => {
-        compileProcessor(schemaSource, useSerializedProcessor)
+  def dynamicLoadTDMLProcessorFactory(className: String): Seq[TDMLDFDLProcessorFactory] = {
+    import scala.language.reflectiveCalls
+    import scala.language.existentials
+
+    val res =
+      try {
+        Seq(Class.forName(className).newInstance().asInstanceOf[TDMLDFDLProcessorFactory])
+      } catch {
+        case cnf: ClassNotFoundException =>
+          System.err.println("TDML Runner did not find implementation '%s'. No tests will be run against it.".format(className))
+          Seq()
+        case ie: InstantiationException => {
+          //
+          // In this case, we found it, but we couldn't create an instance of it, so it's somehow broken,
+          // but there's no point in trying to use it.
+          //
+          System.err.println("TDML Runner found implementation '%s'. But was unable to create an instance. No tests will be run against it.".format(className))
+          Seq()
+        }
+        case cce: ClassCastException => {
+          //
+          // In this case, we found it, created an instance, but it's not the right type.
+          // So there's no point in trying to use it.
+          //
+          System.err.println("TDML Runner found implementation '%s'. But it was not an instance of %s. No tests will be run against it.".format(
+            className, tdmlDFDLProcessorFactoryTraitName
+          ))
+          Seq()
+        }
```

Comment:
Another thought, this could be used to change how we handle debug tests too. We could completely remove the concept of scala-debug, and any tests that do not run under daffodil would just not include daffodil in the list of their implementations. When using daffodil as the implementation, those tests would just be skipped.

Reply:
I like the notion of skipping for tests that aren't targeted.

I can add aliases like ibm-java, dfdl4s-java, etc. 

We need a mechanism to indicate which target we want to test against. 

The bigger issue is fact that you can't switch among implementations without running into classpath  incompatibilities. So I have to combine how to address this issue with that one.

Reply:
This is reimplemented now along the lines of your original comment. No change to scala vs. scala-debug though. This patch set is already super huge.